### ISSUE TITLE
Tests can be overly resource hungry or otherwise get out of control - add hardening somehow?

### ISSUE THREAD

**Issue by Cervator** [Terasology/ModuleTestingEnvironment#30]
Maybe mildly similar to #25 but distinct - in this case `testExample` literally went out of memory:

http://jenkins.terasology.io/teraorg/job/Terasology/job/Modules/job/M/job/ModuleTestingEnvironment/job/PR-29/1/console

I'm not sure what the best way to address this is. It doesn't _look_ like the build agent itself ran out of memory, it had a little more space allocated, although maybe if Java requested a large enough chunk of memory (more than 300MB in one reqest?) that could be related

![image](https://user-images.githubusercontent.com/1063833/100521299-5ce7ff80-3168-11eb-8278-aa1f66558198.png)

Maybe it is something more specific? The build log talks about "GC overhead limit exceeded"

```
DeprecationTest > testExample() STANDARD_ERROR
    Exception in thread "Chunk-Processing-Reactor" java.lang.OutOfMemoryError: GC overhead limit exceeded
    Exception in thread "Thread-148" java.lang.OutOfMemoryError: GC overhead limit exceeded

Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "Chunk-Processing-6"
```

**Comment by DarkWeird** [comment]
I suspect it is not terminated engines during all tests run. Something with shutdowning.
Every client engine takes 600MB+ by chunks (default view distance and server and client sides)

**Comment by Cervator** [comment]
Yeah I tried to add more memory to the build agent but it still failed eventually. On the plus side we now have the watchful http://jenkins.terasology.io/teraorg/job/Utilities/job/BuildWatchinator/ which will find and terminate such stalled builds. So at least they won't stall forever anymore
